### ISSUE TITLE
Filter failure: PRICE_FILTER with try to sell coins

### ISSUE THREAD

**Issue by MyZik** [jaggedsoft/php-binance-api#183]
Hi everyone. I have a code:

```php
$orderBuy = $api->marketBuy('MFTBTC', $notes['quantity']);

if ($orderBuy['code'] !== null) {
    $data['text'] = "⛔️ Buy error:\n\n" .
        "📝 *Text:* `{$orderBuy['msg']}`";
} else {
    $orderSell = $api->sell('MFTBTC', $notes['quantity'], $sellPrice);

    if ($orderSell ['code'] !== null) {
        $data['text'] = "⛔️ Sell error:\n\n" .
            "📝 *Text:* `{$orderSell['msg']}`";
    }
}
```

Variable `$sellPrice` equals MFT price for quantity `$notes['quantity']` and +1.5%

-------
1 MFT Price: 0.00000122 BTC
I tried to buy 820 MFT for  0.00176263 BTC

I use `$orderBuy = $api->marketBuy('MFTBTC', $notes['quantity']);` with no problems. Bought 820 MFT for  0.00176263 BTC.

And try to sell  820 MFT for 0.00101540 BTC  with LIMIT order `$orderSell = $api->sell('MFTBTC', $notes['quantity'], $sellPrice);`. But I've got an error: `Filter failure: PRICE_FILTER`

How can I fix that?

**Comment by jaggedsoft** [comment]
If this does not work, see below for pulling binance minimum order quantities.
```php
$sellPrice = number_format($sellPrice, 8, '.', '');
$orderSell = $api->sell('MFTBTC', $notes['quantity'], $sellPrice);
```
----------

You can use `$api->exchangeInfo();` to pull the minimums for each pair
![example](https://image.ibb.co/bz5KAG/notationals.png)

todo: implement tickSize handling and roundStep like node-binance-api:
#121


@ahab1b here is a script someone made to make it easier on php

https://github.com/jaggedsoft/node-binance-api/issues/23#issuecomment-383795797

```php
$info = $api->exchangeInfo();
$minimums = [];
foreach($info["symbols"] as $symbols){
$filters = ["minNotional"   => "0.001","minQty" => "1","maxQty" => "10000000","stepSize"   => "1","minPrice" => "0.00000001","maxPrice"   => "100000"];
$filters["orderTypes"] = implode(",",$symbols["orderTypes"]);
	foreach($symbols["filters"] as $filter){
		if ( $filter["filterType"] == "MIN_NOTIONAL" ) {
				$filters["minNotional"] = $filter["minNotional"];
			} else if ( $filter["filterType"] == "PRICE_FILTER" ) {
				$filters["minPrice"] = $filter["minPrice"];
				$filters["maxPrice"] = $filter["maxPrice"];
			} else if ( $filter["filterType"] == "LOT_SIZE" ) {
				$filters["minQty"] = $filter["minQty"];
				$filters["maxQty"] = $filter["maxQty"];
				$filters["stepSize"] = $filter["stepSize"];
			}
		}
$minimums[$symbols["symbol"]] = $filters;
}
$fp = fopen('exchangeinfo.json', 'w');
fwrite($fp, json_encode($minimums));
fclose($fp);
```

**Comment by MyZik** [comment]
@jaggedsoft this solution doesn't help me :(
```php
$sellPrice = number_format($sellPrice, 8, '.', '');
$orderSell = $api->sell('MFTBTC', $notes['quantity'], $sellPrice);
```

I tried to show `$api->exchangeInfo();` with `var_export()`, but there is no way for symbol 'MFTBTC'...

**Comment by jaggedsoft** [comment]
Says here you must order a minimum of 1 MFT with no decimal places. So you can only order 1 MFT, 2 MFT, etc.

```js
{
        {
            "symbol": "MFTBTC",
            "status": "TRADING",
            "baseAsset": "MFT",
            "baseAssetPrecision": 8,
            "quoteAsset": "BTC",
            "quotePrecision": 8,
            "orderTypes": [
                "LIMIT",
                "LIMIT_MAKER",
                "MARKET",
                "STOP_LOSS_LIMIT",
                "TAKE_PROFIT_LIMIT"
            ],
            "icebergAllowed": true,
            "filters": [
                {
                    "filterType": "PRICE_FILTER",
                    "minPrice": "0.00000017",
                    "maxPrice": "0.00001640",
                    "tickSize": "0.00000001"
                },
                {
                    "filterType": "LOT_SIZE",
                    "minQty": "1.00000000",
                    "maxQty": "90000000.00000000",
                    "stepSize": "1.00000000"
                },
                {
                    "filterType": "MIN_NOTIONAL",
                    "minNotional": "0.00100000"
                },
                {
                    "filterType": "ICEBERG_PARTS",
                    "limit": 10
                },
                {
                    "filterType": "MAX_NUM_ALGO_ORDERS",
                    "maxNumAlgoOrders": 5
                }
            ]
        }
}
```

**Comment by MyZik** [comment]
@jaggedsoft I bought (MARKET BUY) 819 MFT for 0.00099917 BTC and am trying to sell 819 MFT for 0.00101415 BTC at once, after market buy. What's wrong? I don't violate the limits...

**Comment by jaggedsoft** [comment]
MIN_NOTIONAL is 0.001, meaning that you must buy or sell at minimum .001 BTC worth at a time. Sometimes market orders don't have these minimums, but limit orders always do.

**Comment by MyZik** [comment]
@jaggedsoft tried to buy 840 MFT for 0.00101640 BTC and it's done. But selling for 0.00103164 BTC with error `Filter failure: PRICE_FILTER` again :(
And 0.00103164 bigger than 0.001...

**Comment by ildarius** [comment]
When you sent the order binance returns output, can you copy/paste it? and
also your input (whatever you had in $orderBuy)

also you can scrape price/quantity filters directly from
https://api.binance.com/api/v1/exchangeInfo

**Comment by jaggedsoft** [comment]
Available in v0.4.5
`$quantity = $api->roundStep($quantity, $stepSize);`

```php
function roundStep($value, $stepSize = 0.1) {
        $precision = strlen(substr(strrchr(rtrim($value,'0'), '.'), 1));
        return round((($value / $stepSize) | 0) * $stepSize, $precision);
}
```

**Comment by farshadth** [comment]
hi bro
i`m using your roundStep() method to submit sell order but i still get price filter error 
and only works for bitcoin
help me bro tnx
